js-core/homeworks/homework-15/src/main.js

```diff
@@ -0,0 +1,386 @@
+    return Array.isArray(elem) ? newArray.concat(flatten(elem)) : newArray.concat(elem);
+  }, []);
+};
+
+console.log(flatten([[1,2],[3,[4]],5, 10]));
+console.log(flatten([25, 10, [10, [15]]]));
+console.log(flatten([1, [2, [ {a: "b", c: 'd' }, { c: [1, 2, 5] } ] ] ]));
+
+
+/*
+Виртуализировать таблицу, сделать рендер всей
+таблицы через JavaScript.
+Второй макет.
+https://github.com/aleksandra-maslennikova/telephone-book/blob/master/index.html
+Выглядеть должно так же: https://aleksandra-maslennikova.github.io/telephone-book/index.html
+*/
+
+const contentForThead = ['Name', 'Last Name', 'Email'];
+
+const users = [
+    [
+        {
+            name: 'Ivan',
+            lastName: 'Ivanov',
+            email: 'ivan-ivanov@vkontakte.com'
+        }
+    ],
+    [
+        {
+            name: 'Ivan',
+            lastName: 'Ivanov',
+            email: 'ivan-ivanov@vkontakte.com'
+        }
+    ],
+    [
+        {
+            name: 'Ivan',
+            lastName: 'Ivanov',
+            email: 'ivan-ivanov@vkontakte.com'
+        }
+    ],
+    [
+        {
+            name: 'Ivan',
+            lastName: 'Ivanov',
+            email: 'ivan-ivanov@vkontakte.com'
+        }
+    ],
+    [
+        {
+            name: 'Ivan',
+            lastName: 'Ivanov',
+            email: 'ivan-ivanov@vkontakte.com'
+        }
+    ],
+    [
+        {
+            name: 'Ivan',
+            lastName: 'Ivanov',
+            email: 'ivan-ivanov@vkontakte.com'
+        }
+    ],
+    [
+        {
+            name: 'Ivan',
+            lastName: 'Ivanov',
+            email: 'ivan-ivanov@vkontakte.com'
+        }
+    ],
+    [
+        {
+            name: 'Ivan',
+            lastName: 'Ivanov',
+            email: 'ivan-ivanov@vkontakte.com'
+        }
+    ],
+    [
+        {
+            name: 'Ivan',
+            lastName: 'Ivanov',
+            email: 'ivan-ivanov@vkontakte.com'
+        }
+    ],
+    [
+        {
+            name: 'Ivan',
+            lastName: 'Ivanov',
+            email: 'ivan-ivanov@vkontakte.com'
+        }
+    ],
+    [
+        {
+            name: 'Ivan',
+            lastName: 'Ivanov',
+            email: 'ivan-ivanov@vkontakte.com'
+        }
+    ],
+    [
+        {
+            name: 'Ivan',
+            lastName: 'Ivanov',
+            email: 'ivan-ivanov@vkontakte.com'
+        }
+    ],
+    [
+        {
+            name: 'Ivan',
+            lastName: 'Ivanov',
+            email: 'ivan-ivanov@vkontakte.com'
+        }
+    ],
+    [
+        {
+            name: 'Ivan',
+            lastName: 'Ivanov',
+            email: 'ivan-ivanov@vkontakte.com'
+        }
+    ],
+    [
+        {
+            name: 'Ivan',
+            lastName: 'Ivanov',
+            email: 'ivan-ivanov@vkontakte.com'
+        }
+    ],
+    [
+        {
+            name: 'Ivan',
+            lastName: 'Ivanov',
+            email: 'ivan-ivanov@vkontakte.com'
+        }
+    ],
+    
+    
+];
+
+const footerTabs = [
```

Comment:
it's not the data, it just a static part of the content and it's not required to virtualize it